packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -388,7 +388,9 @@ export class ApolloServerBase {
           // removed until after the rest of shutdown happens.
           process.kill(process.pid, signal);
         };
-        process.on(signal, handler);
+        if (!process.listeners(signal).some((listener: Function) =>  listener.toString() === handler.toString())) {
```

Comment:
I'm not sure this is good enough — I think we want to compare not just "is it an apollo server handler" but "is it an apollo server handler for *this* ApolloServer". Is there a nice more-or-less-typesafe way to make the handler have some kind of field that lets us identify it as (a) this handler and (b) pointing to this particular AS?

Reply:
Or hmm, I guess this is the constructor, so it's always a different ApolloServer. I'm not sure why you're making a new ApolloServer for each request within the same runtime, and never `stop`ping them? That's gonna cause more problems than just too many signal handlers.  Which serverless architecture are you in and how are you setting up your server?

Reply:
@glasser I'm using [`aws-lambda-graqphl`](https://github.com/michalkvasnicak/aws-lambda-graphql) with the setup specified in its README, which depends on `apollo-server-lambda` internally but adds functionality for GraphQL subscriptions.

You raise a good point about starting up a new ApolloServer for each request, without the proper cleanup. I would think that, even in a serverless model, the resources from the last lambda invocation should be cleaned up (and the server instance `stop`ped) before the next iteration.

After doing some more research, I think the problem might be that the ApolloServer `stop` event is only called on `SIGTERM` and `SIGINT` events, whereas AWS Lambda does not send these signals when a function invocation is finished.

This doesn't seem like a big problem for production, since Lambda environments automatically get destroyed. I do think there's a potential fix here for myself and other `serverless-offline` users, but it might involve changes to the way ApolloServer is initialized in `aws-lambda-graphql` specifically, so I could just close this PR and open an issue there instead?